pkg/network/server.go

```diff
@@ -331,12 +334,33 @@ func (s *Server) startProtocol(p Peer) {
 			err = p.WriteMsg(m)
 		case <-timer.C:
 			// Try to sync in headers and block with the peer if his block height is higher then ours.
-			if p.Version().StartHeight > s.chain.BlockHeight() {
+			if p.LastBlockIndex() > s.chain.BlockHeight() {
 				err = s.requestBlocks(p)
 			}
 			if err == nil {
 				timer.Reset(s.ProtoTickInterval)
 			}
+			if s.chain.HeaderHeight() >= p.LastBlockIndex() {
+				block, err := s.chain.GetBlock(s.chain.CurrentBlockHash())
+				if err != nil {
+					return
+				}
+				diff := uint32(time.Now().UTC().Unix()) - block.Timestamp
+				if diff > uint32(s.PingInterval/time.Second) {
+					if err := p.WriteMsg(NewMessage(s.Net, CMDPing, payload.NewPing(s.id, s.chain.HeaderHeight()))); err != nil {
+						return
+					}
+					s.pingSent = true
+				}
+			}
+		case <-pingTimer.C:
+			if s.pingSent {
+				s.unregister <- peerDrop{p, errors.New("ping/pong timeout")}
```

Comment:
Just set an error an it will be done in error handler down below.

Reply:
fixed